File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        traverse(root);
        return res;
    }
  
    List<Integer> res = new LinkedList<>();
    
    void traverse(Node root) {
      if(root == null) return;
      for(Node child: root.children) 
        traverse(child);
      
      res.add(root.val);
    }
}

File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<Node*> stack;
    vector<int> postorder(Node* root) {
        if(!root) return {};
        vector<int> ans;
        Node * t;
      
        stack.push_back(root);
        for (Node* e: stack) cout<< e<<endl;
        
        while(!stack.empty()) {
          t = stack.back();
          stack.pop_back();
          
          ans.push_back(t->val);
          for (Node* child: t->children)
            stack.push_back(child);
        }
        
        reverse(ans.begin(), ans.end());
      
        return ans;
    }
};